### DISCUSSION TITLE
How to adjust the size of graph.

### DISCUSSION THREAD

**Discussion by peacerider** [kernc/backtesting.py#1021] [Q&A] [answered]
Hello,
I am now learning backtesting.py.
Could someone tell me how to change the size (height) of the graph.

Example
I want to adjust the height of the MACD figure below.
![20230715_MACD](https://github.com/kernc/backtesting.py/assets/99186617/bcaaaccb-83fa-4469-bc4b-ca114c11dadf)

I wached the comment in the source code.
"The height is currently non-adjustable."
https://github.com/kernc/backtesting.py/blob/master/backtesting/backtesting.py

**Comment by peacerider** [comment]
I changed the parameter of the source code directry.

**Comment by Rydo111** [comment, marked as answer]
@peacerider is correct. If you look at `_plotting.py`, you'll find the height settings of each section. For example, the main chart height is configured on line 221 (`height=400`). You could manually change this to whatever you want; or, I added a parameter to `def plot` on line 176 that I then passed to `new_bokeh_figure`. You can do the same for `new_indicator_figure` on line 277.

Hope this helps.

**Reply by peacerider** [reply]
Thank you for your help, especially about the second half of the advice!

**Reply by Rydo111** [reply]
No worries. The plotting code is quite good, and very useful by itself outside of the utility of backtesting. I'd like to contribute to the code if I can...I'm a bit of a noob when it comes to Git (and to coding in general tbh), so I don't really know how the whole process works (forking, push request, etc.). I'll do some study and see if I can make some suggestions directly into the code.

**Reply by peacerider** [reply]
Thanks, I am also beginer of Git.

**Reply by Rydo111** [reply]
Hey @peacerider, for your interest, I managed to fork the repo, integrate this into my pycharm and now I am pushing commits to my repo. Its like magic! ;). 

I've made some code changes you may be interested in. My repo is https://github.com/Rydo111/backtesting.py.git. Things I've played with:
* created a hook in plot to control the line thickness of indicators
* created a relative volume segment, that sits above the OHLC plot and dynamically re-scales when the x-axis changes.
* added a parameter called `ohlc_height` to `plot` to gain greater control of the OHLC plot segment

I'm going to keep messing about...I don't think I'll try to do any push requests...my code isn't really good enough.

**Reply by sanketgarade** [reply]
here is a simple patch (to override the existing height parameter) that you can put in your own script where you run `bt.plot()`
```
# override existing indicator plot heights ---- start
import backtesting._plotting as bt_plotting

# Override the default indicator height globally
bt_plotting._INDICATOR_HEIGHT = 200
# override existing indicator plot heights ---- end
```
